vnext/Microsoft.ReactNative/Utils/Helpers.cpp

```diff
@@ -95,8 +96,13 @@ bool Is19H1OrHigher() {
 }
 
 bool IsXamlIsland() {
+  auto appModel = managed_resource(
+      LoadLibraryEx(L"Api-ms-win-appmodel-runtime-l1-1-2.dll", 0, LOAD_LIBRARY_SEARCH_SYSTEM32), FreeLibrary);
+  decltype(&AppPolicyGetWindowingModel) pfnAppPolicyGetWindowingModel =
+      reinterpret_cast<decltype(&AppPolicyGetWindowingModel)>(GetProcAddress(appModel, "AppPolicyGetWindowingModel"));
+
   AppPolicyWindowingModel e;
-  if (FAILED(AppPolicyGetWindowingModel(GetCurrentThreadEffectiveToken(), &e)) ||
+  if (FAILED(pfnAppPolicyGetWindowingModel(GetCurrentThreadEffectiveToken(), &e)) ||
       e == AppPolicyWindowingModel_ClassicDesktop) {
     return true;
```

Comment:
Consider caching the answer to this since we made this a bit more expensive.

Reply:
@NickGerleman  the problem is this can change over time (technically an app could create an island, tear it down, then run a xaml app on the same thread, or vice versa)